src/java.naming/share/classes/com/sun/jndi/ldap/LdapPoolManager.java

```diff
@@ -395,47 +395,21 @@ private static void d(String msg, String o) {
         }
     }
 
-    private static final String getProperty(final String propName,
-        final String defVal) {
+    private static final String getProperty(final String propName, final String defVal) {
         return AccessController.doPrivileged(
-            new PrivilegedAction<String>() {
-            public String run() {
-                try {
-                    return System.getProperty(propName, defVal);
-                } catch (SecurityException e) {
-                    return defVal;
-                }
-            }
-        });
+                (PrivilegedAction<String>) () -> System.getProperty(propName, defVal)
+        );
```

Comment:
If you want to avoid the cast then you could create the PrivilegedAction explicitly, e.g.

PrivilegedAction<String> pa = () -> System.getProperty(propName, defVal);
return AccessController.doPrivileged(pa);

Reply:
An alternative here would be to use sun.security.action.privilegedGetProperty(prop, default).
The package is already exported from java.base to java.desktop, etc.

Reply:
That is a very neat alternative yes. Approaching the problem like that especially improves the readability [JdkLDAP.java](https://github.com/openjdk/jdk/pull/3416/files#diff-bf4c67da93cf2b9196508db2d57f7e01bc884f2268f5bfd43a9f01dfd55e4955) in my opinion. 
I don't think casting has any major performance hits for these fixes, so is your suggestion mostly for readability's sake?

Reply:
Right, not a performance problem.  Just simpler to use an existing method to read a privileged property.
And there will be one less doPriv.

Reply:
I think I may change the PR to reflect what Alan suggested. Its more readable than the lambda-with-cast solution in that an action is created _and then_ an action is carried out as supposed to fitting it all into one call. Its also as concise.